python/src/pycryptosat.cpp

```diff
@@ -348,11 +348,15 @@ Add iterable of clauses to the solver.\n\
 
 static PyObject* add_clauses(Solver *self, PyObject *args, PyObject *kwds)
 {
-    static char* kwlist[] = {"clauses", NULL};
+    static char* kwlist[] = {"clauses", "max_var", NULL};
     PyObject *clauses;
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O", kwlist, &clauses)) {
+    long int max_var = 0;
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|l", kwlist, &clauses, &max_var)) {
         return NULL;
     }
+    if (max_var >= (long int)self->cmsat->nVars()) {
```

Comment:
I simply copied that logic from https://github.com/msoos/cryptominisat/commit/3a2b9a60e02e061df322a4664ce119f8c449bd75#diff-437bd5e874df2ba9f9ced39b13eee9abR185.
I'm a little unsure about the offset, though, as I'm too unfamiliar with CMS inner workings, e.g., in this case when zero- or one-based variable indices are used.
If we have `max_var == 0`, this will add a new variable, which seems odd. Is there always a base variable, i.e., is `self->cmsat->nVars() >= 1` always true?

Reply:
This needs to be changed to `>` instead of `>=`. Here, the max_var is the actual variable number, not zero-starting.

Reply:
I see, so I guess in the part I copied from, the offset is different because of the `convert_lit_to_sign_and_var` call at https://github.com/msoos/cryptominisat/commit/3a2b9a60e02e061df322a4664ce119f8c449bd75#diff-437bd5e874df2ba9f9ced39b13eee9abL173.